### ISSUE TITLE
`Naming/MethodName` doesn't respect operator methods defined with `define_singleton_method`

### ISSUE THREAD

**Issue by sampart** [rubocop/rubocop#14382]
## Expected behavior

Given the following file:

```ruby
define_singleton_method(:[])
define_singleton_method(:+)
```

I would expect `bin/rubocop --only=Naming/MethodName` to report no errors.

## Actual behavior

Two violations are reported:

```text
bin/rubocop --debug --only=Naming/MethodName temp.rb
For /path/redacted: configuration from /path/redacted/.rubocop.yml
configuration from /path/redacted/vendor/gems/3.4.5/ruby/3.4.0/gems/rubocop-rails-accessibility-1.0.1/config/default.yml
configuration from /path/redacted/vendor/gems/3.4.5/ruby/3.4.0/gems/rubocop-rails-accessibility-1.0.1/config/default.yml
Default configuration from /path/redacted/vendor/gems/3.4.5/ruby/3.4.0/gems/rubocop-1.78.0/config/default.yml
Inheriting configuration from /path/redacted/vendor/gems/3.4.5/ruby/3.4.0/gems/primer_view_components-0.43.4/lib/rubocop/config/default.yml
Inheriting configuration from /path/redacted/vendor/gems/3.4.5/ruby/3.4.0/gems/rubocop-sorbet-0.10.5/config/default.yml
Inheriting configuration from /path/redacted/vendor/gems/3.4.5/ruby/3.4.0/gems/rubocop-github-0.26.0/config/default.yml
Plugin configuration from /path/redacted/vendor/gems/3.4.5/ruby/3.4.0/gems/rubocop-performance-1.24.0/config/default.yml
configuration from /path/redacted/vendor/gems/3.4.5/ruby/3.4.0/gems/rubocop-github-0.26.0/config/default_cops.yml
configuration from /path/redacted/vendor/gems/3.4.5/ruby/3.4.0/gems/rubocop-github-0.26.0/config/default_cops.yml
Inheriting configuration from /path/redacted/vendor/gems/3.4.5/ruby/3.4.0/gems/rubocop-github-0.26.0/config/default_pending.yml
Inheriting configuration from /path/redacted/vendor/gems/3.4.5/ruby/3.4.0/gems/rubocop-github-0.26.0/config/rails.yml
Plugin configuration from /path/redacted/vendor/gems/3.4.5/ruby/3.4.0/gems/rubocop-rails-2.30.3/config/default.yml
configuration from /path/redacted/vendor/gems/3.4.5/ruby/3.4.0/gems/rubocop-github-0.26.0/config/rails_cops.yml
configuration from /path/redacted/vendor/gems/3.4.5/ruby/3.4.0/gems/rubocop-github-0.26.0/config/rails_cops.yml
Inheriting configuration from /path/redacted/vendor/gems/3.4.5/ruby/3.4.0/gems/rubocop-github-0.26.0/config/rails_pending.yml
Inheriting configuration from /path/redacted/.rubocop_todo.yml
Inheriting configuration from /path/redacted/.rubocop_axe_scan.yml
Inheriting configuration from /path/redacted/.rubocop_monolith_db.yml
Inspecting 1 file
Scanning /path/redacted/temp.rb
C

Offenses:

temp.rb:1:25: C: Naming/MethodName: Use snake_case for method names. (https://rubystyle.guide#snake-case-symbols-methods-vars)
define_singleton_method(:[])
                        ^^^^
temp.rb:2:25: C: Naming/MethodName: Use snake_case for method names. (https://rubystyle.guide#snake-case-symbols-methods-vars)
define_singleton_method(:+)
                        ^^^

1 file inspected, 2 offenses detected
Finished in 1.2340044069999294 seconds
```

## Steps to reproduce the problem

Create a file which calls `define_singleton_method` with an operator method. Here's an example:

```ruby
define_singleton_method(:[])
define_singleton_method(:+)
```

Run `bin/rubocop --only=Naming/MethodName` on it.

## RuboCop version

```text
$ bin/rubocop -V
1.78.0 (using Parser 3.3.8.0, Prism 1.4.0, rubocop-ast 1.45.1, analyzing as Ruby 3.3, running on ruby 3.4.5) [x86_64-linux]
  - rubocop-performance 1.24.0
  - rubocop-rails 2.30.3
  - rubocop-sorbet 0.10.5
```

**Comment by viralpraxis** [comment]
It was fixed by https://github.com/rubocop/rubocop/pull/14351 (which is not released yet)

**Comment by sampart** [comment]
Aha, thank you! (I did look for an issue but couldn't find one.) ❤️
